src/jit/morph.cpp

```diff
@@ -7411,6 +7411,41 @@ void Compiler::fgMorphTailCallViaHelper(GenTreeCall* call, void* pfnCopyArgs)
     if (call->IsVirtualStub())
     {
         call->gtFlags |= GTF_CALL_NULLCHECK;
+
+#if defined(_TARGET_AMD64_)
+        // If we already inited arg info here then we will have added the VSD
+        // arg on AMD64. So we remove it here as we will handle this case
+        // specially below.
+        fgArgInfo* argInfo = call->fgArgInfo;
+        if (argInfo != nullptr)
+        {
+            GenTreeCall::Use* vsdArg = nullptr;
+
+            for (unsigned index = 0; index < argInfo->ArgCount(); ++index)
+            {
+                fgArgTabEntry* arg = argInfo->GetArgEntry(index, false);
+                if (arg->isNonStandard)
+                {
+                    // The only supported nonstandard arg for slow tailcalls is
+                    // VSD arg.
+                    assert(vsdArg == nullptr);
+                    vsdArg = arg->use;
+#ifndef DEBUG
+                    break;
+#endif
+                }
+            }
+
+            assert(vsdArg != nullptr);
```

Comment:
This seems fine to me but I think it warrants its own comment. It took a little to reason what this was doing and why.

Reply:
I suppose you mean the loop below this line?